File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.py
class Solution:
    def smallestIndex(self, num: list[int]) -> int:
        for i, n in enumerate(num):
            temp = n
            digit_sum = 0
            
            while temp > 0:
                digit_sum += temp % 10
                temp //= 10
                
            if i == digit_sum:
                return i
                
        return -1

File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.java
class Solution {
    public int smallestIndex(int[] num) {
        for (int i = 0; i < num.length; i++) {
            int sum = 0;
            int temp = num[i]; 
            while (temp > 0) {
                sum += temp % 10;
                temp /= 10;
            } 
            if (i == sum) {
                return i;
            }
        } 
        return -1;
    }
}